alpha/apps/kaltura/modules/extwidget/actions/playManifestAction.class.php

```diff
@@ -1296,7 +1296,21 @@ public function execute()
 		$renderer->setKsObject(kCurrentContext::$ks_object);
 		$renderer->setPlaybackContext($playbackContext);
 		$renderer->setDeliveryCode($deliveryCode);
-		
+
+		if ($this->secureEntryHelper->getExtraPropertyByName('ServeFromKESRule'))
+		{
+			$playLocation = 'external';
+			if ($this->deliveryAttributes->getEdgeServerIds() && count($this->deliveryAttributes->getEdgeServerIds()))
+			{
+				$playLocation = implode(",", $this->deliveryAttributes->getEdgeServerIds());
+				header('X-ServerNodeId:' . $playLocation);
```

Comment:
X-ServerNodeId: --> X-ServerNodeIds:
getEdgeServerIds fro live will return empty array at this step + the edge server ids actually used is only known inside the delivery Profile so you need to somehow fetch it from there